crates/subspace-farmer/src/single_plot_farm.rs

```diff
@@ -300,17 +300,16 @@ pub struct PlotFactoryOptions<'a> {
     pub max_plot_size: u64,
 }
 
-pub trait PlotFactory =
-    Fn(PlotFactoryOptions<'_>) -> Result<Plot, PlotError> + Send + Sync + 'static;
+pub trait PlotFactory = Fn(PlotFactoryOptions<'_>) -> Result<Plot, PlotError> + Send + Sync;
 
-pub(crate) struct SinglePlotFarmOptions<'a, RC, PF> {
+pub(crate) struct SinglePlotFarmOptions<RC, PF> {
     pub(crate) id: SinglePlotFarmId,
     pub(crate) plot_directory: PathBuf,
     pub(crate) metadata_directory: PathBuf,
     pub(crate) allocated_plotting_space: u64,
     pub(crate) farmer_protocol_info: FarmerProtocolInfo,
     pub(crate) farming_client: RC,
-    pub(crate) plot_factory: &'a PF,
+    pub(crate) plot_factory: PF,
```

Comment:
I'm not necessarily strongly opposed to this change, but why was it necessary?

Reply:
Hmm, I was just tweaking around and found out that having `&impl Fn` is actually useless as `Fn` is implemented for `&impl Fn`.
So I just decided to commit that :)

Reply:
Yes, but the lifetime now has to be `'static`. But if nothing took advantage of that, I guess we're okay either way.

Reply:
No, it doesn't have to be actually. Everything remained the same in terms of implementation. The api was relaxed:
- `PlotFactory` now doesn't require `'static` lifetime (we don't need it for `SinglePlotFarming`, as we used reference to it before)
- Now `SinglePlotFarming` accepts both references to `PlotFactory` and `PlotFactory` itself

Reply:
`SingleDiskFarm::new()` now takes `PlotFactory + 'static` as an argument. Can probably go around it by using scoped threads from standard library instead of `tokio::task::spawn_blocking` and entering context manually. Though may not be worth it.

Reply:
No, actually beforehand `PlotFactory: 'static`, so `PlotFactory` was weakened, while `SingleDiskFarm::new` requirements stayed the same. :man_shrugging:

Reply:
Ah, I see, makes sense :+1: